
From 6685a0621c0d1136690826f6366c96b3d28c9587 Mon Sep 17 00:00:00 2001
From: Balaji Selvam <balaji_selvam@comcast.com>
Date: Wed, 12 Aug 2020 11:00:43 +0000
Source: Metrological
Upstream-Status: Pending https://github.com/anholt/libepoxy/pull/144
Date: Sat, 24 Nov 2018 20:03:05 -0800
Subject: [PATCH] Make graphic libs configurable
Co-authored-by: Bram Oosterhuis <b.oosterhuis@gmail.com>
Signed-off-by: wouterlucas <wouter@wouterlucas.com>

Recreated by: thulasiram_kadiyala <thulasiram_kadiyala@comcast.com>
Recreated from : https://gerrit.teamccp.com/plugins/gitiles/rdk/yocto_oe/layers/meta-rdk-ext/+/topic/fr_yocto4.0/recipes-graphics/libepoxy/libepoxy/0004-Make-graphic-libs-configurable_v1.5.4.patch

Index: libepoxy-1.5.9/src/dispatch_common.c
===================================================================
--- libepoxy-1.5.9.orig/src/dispatch_common.c
+++ libepoxy-1.5.9/src/dispatch_common.c
@@ -173,6 +173,9 @@
 
 #include "dispatch_common.h"
 
+#define xstr(a) str(a)
+#define str(a) #a
+
 #if defined(__APPLE__)
 #define GLX_LIB "/opt/X11/lib/libGL.1.dylib"
 #define OPENGL_LIB "/System/Library/Frameworks/OpenGL.framework/Versions/Current/OpenGL"
@@ -190,10 +193,10 @@
 #define OPENGL_LIB "OPENGL32"
 #else
 #define GLVND_GLX_LIB "libGLX.so.1"
-#define GLX_LIB "libGL.so.1"
-#define EGL_LIB "libEGL.so.1"
-#define GLES1_LIB "libGLESv1_CM.so.1"
-#define GLES2_LIB "libGLESv2.so.2"
+#define GLX_LIB  xstr(GLX_LIB_NAME)
+#define EGL_LIB xstr(EGL_LIB_NAME)
+#define GLES1_LIB xstr(GLES1_LIB_NAME)
+#define GLES2_LIB xstr(GLES2_LIB_NAME)
 #define OPENGL_LIB "libOpenGL.so.0"
 #endif
 
